File: elementary-graph-algorithms/python/graph.py
import enum


class Color(enum.Enum):
    WHITE = "white"
    GRAY = "gray"
    BLACK = "black"


class Vertex:
    def __init__(self, name):
        self.name = name
        self.distance = 0
        self.predecessor = None
        self.color = Color.WHITE


class Edge:
    def __init__(self, start, end, weight=1):
        self.start = start
        self.end = end
        self.weight = weight


class Graph:
    def __init__(self, vertices, edges):
        self.vertices = vertices
        self.edges = edges
        self.adjacency_list = {}
        self.__initialize_graph()

    def print_graph(self):
        for k in self.vertices.keys():
            print(k, end=" : ")
            adjacencies = self.adjacency_list[k]
            for i, adjacency in enumerate(adjacencies):
                print(adjacency.name, end=" -> " if i < len(adjacencies) - 1 else "")

            print("")

    def __initialize_graph(self):
        for k, v in self.vertices.items():
            self.adjacency_list[k] = []

        for edge in self.edges:
            self.adjacency_list[edge.start].append(self.vertices[edge.end])

File: elementary-graph-algorithms/golang/graph.go
package main

import "fmt"

type Vertex struct {
	name                                  string
	predecessor                           *Vertex
	distance, start_time, end_time, color int
}

type Edge struct {
	start, end string
}

type Graph struct {
	vertices      map[string]*Vertex
	edges         []*Edge
	adjacencyList map[string][]*Vertex
}

func (graph *Graph) initializeGraph() {
	graph.adjacencyList = make(map[string][]*Vertex)
	for _, v := range graph.vertices {
		graph.adjacencyList[v.name] = make([]*Vertex, 0)
	}
	for _, edge := range graph.edges {
		graph.adjacencyList[edge.start] = append(graph.adjacencyList[edge.start], graph.vertices[edge.end])
	}
}

func (graph *Graph) printGraph() {
	for _, v := range graph.vertices {
		fmt.Print(v.name, " : ")

		adjacencies := graph.adjacencyList[v.name]
		for i, adjacency := range adjacencies {
			fmt.Printf("%v", adjacency.name)
			if i < len(adjacencies)-1 {
				fmt.Print(" -> ")
			}
		}

		fmt.Print("\n")
	}
}
